staging/src/k8s.io/cloud-provider/controllers/node/node_controller_test.go

```diff
@@ -675,6 +682,517 @@ func Test_AddCloudNode(t *testing.T) {
 				},
 			},
 		},
+		//{
+		//	name: "[instanceV2] node initialized with provider ID",
+		//	fakeCloud: &fakecloud.Cloud{
+		//		EnableInstancesV2: true,
+		//		InstanceTypes: map[types.NodeName]string{
+		//			types.NodeName("node0"): "t1.micro",
+		//		},
+		//		ExtID: map[types.NodeName]string{
+		//			types.NodeName("node0"): "12345",
+		//		},
+		//		Addresses: []v1.NodeAddress{
+		//			{
+		//				Type:    v1.NodeHostName,
+		//				Address: "node0.cloud.internal",
+		//			},
+		//			{
+		//				Type:    v1.NodeInternalIP,
+		//				Address: "10.0.0.1",
+		//			},
+		//			{
+		//				Type:    v1.NodeExternalIP,
+		//				Address: "132.143.154.163",
+		//			},
+		//		},
+		//		ErrByProviderID: nil,
+		//		Err:             nil,
+		//	},
+		//	existingNode: &v1.Node{
+		//		ObjectMeta: metav1.ObjectMeta{
+		//			Name:              "node0",
+		//			CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+		//		},
+		//		Spec: v1.NodeSpec{
+		//			Taints: []v1.Taint{
+		//				{
+		//					Key:    cloudproviderapi.TaintExternalCloudProvider,
+		//					Value:  "true",
+		//					Effect: v1.TaintEffectNoSchedule,
+		//				},
+		//			},
+		//		},
+		//	},
+		//	updatedNode: &v1.Node{
+		//		ObjectMeta: metav1.ObjectMeta{
+		//			Name:              "node0",
+		//			CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+		//		},
+		//		Spec: v1.NodeSpec{
+		//			ProviderID: "fake://12345",
+		//		},
+		//	},
+		//},
+		{
+			name: "[instanceV2] node ignored",
+			fakeCloud: &fakecloud.Cloud{
+				EnableInstancesV2: true,
+				InstanceTypes: map[types.NodeName]string{
+					types.NodeName("node0"):        "t1.micro",
+					types.NodeName("fake://12345"): "t1.micro",
+				},
+				ExtID: map[types.NodeName]string{
+					types.NodeName("node0"): "12345",
+				},
+				Addresses: []v1.NodeAddress{
+					{
+						Type:    v1.NodeHostName,
+						Address: "node0.cloud.internal",
+					},
+					{
+						Type:    v1.NodeInternalIP,
+						Address: "10.0.0.1",
+					},
+					{
+						Type:    v1.NodeExternalIP,
+						Address: "132.143.154.163",
+					},
+				},
+				Err: nil,
+			},
+			existingNode: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "node0",
+					CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			updatedNode: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "node0",
+					CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+		},
+		//{
+		//	name: "[instanceV2] zone/region topology labels added",
```

Comment:
Are these TODO tests since the fake cloud provider doesn't implement v2?

Reply:
Reason is https://github.com/kubernetes/kubernetes/pull/91319#issuecomment-647309227

Reply:
Not sure whether to change expected node to make it different from current tests which use instances.